js/service/shell/index.js

```diff
@@ -44,3 +44,20 @@ exports.setCommand = function(name, cb) {
   assert(typeutils.isFunction(cb));
   commands.set(name, cb);
 };
+
+exports.runCommand = function(name, args, cb) {
+  var capture = require('./captureio.js');
+
+  assert(typeutils.isString(name));
+
+  args = args || "";
+
+  commands.get(name)(args, capture.io, function() { capture.events.emit('end'); });
+
+  if (cb) {
+    capture.events.redrawPrompt = inputBox.done;
+    cb(capture.events);
+  }
```

Comment:
1: The callback is `socket.io` style, where it gives you an EventEmitter that allows you emit events and receive events.
2: The callback isn't required, you probably saw that it returns the EE as well, so it can be set to a variable and used from there.